os/include/tinyara/mipidsi/mipi_dsi.h

```diff
@@ -0,0 +1,864 @@
+/****************************************************************************
+ *
+ * Copyright 2024 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * include/nuttx/video/mipi_dsi.h
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+#ifndef __INCLUDE_NUTTX_VIDEO_MIPI_DSI_H
+#define __INCLUDE_NUTTX_VIDEO_MIPI_DSI_H
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <tinyara/fs/ioctl.h>
+
+#include <stdbool.h>
+#include <stddef.h>
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+/* Define cmd for dsi device */
+
+#define MIPIDSI_GETDEVLANES     _MIPIDSIIOC(1)
+#define MIPIDSI_GETDEVFMT       _MIPIDSIIOC(2)
+#define MIPIDSI_GETDEVMODE      _MIPIDSIIOC(3)
+#define MIPIDSI_GETDEVHSRATE    _MIPIDSIIOC(4)
+#define MIPIDSI_GETDEVLPRATE    _MIPIDSIIOC(5)
+
+/* Define cmd for dsi host */
+
+#define MIPIDSI_TRANSFER        _MIPIDSIIOC(6)
+
+/* Define msg flags */
+
+#define MIPI_DSI_MSG_REQ_ACK      (1 << 0)	/* Request ACK from peripheral */
+#define MIPI_DSI_MSG_USE_LPM      (1 << 1)	/* Use Low Power Mode to
+											 * transmit message */
```

Comment:
This file is having many misalignment, could you please check

Reply:
I ran the cformatter on all files as given in wiki page.
Didn't knew it make the code violate coding rules.

Reply:
I ran the cformatter on all files as given in wiki page.
Didn't knew it make the code violate coding rules.
I will fix this manually.